taskiq/depends/progress_tracker.py

```diff
@@ -0,0 +1,99 @@
+import enum
+from typing import Generic, Union, overload
+
+from pydantic.generics import GenericModel
+from taskiq_dependencies import Depends
+from typing_extensions import TypeVar
+
+from taskiq.context import Context
+
+# TODO: PEP 696
+# _ProgressType = TypeVar("_ProgressType", default=Any) #  noqa: E800
+_ProgressType = TypeVar("_ProgressType")
+
+
+class Undefined(enum.Enum):
+    """For undefined value."""
+
+    undefined = ...
+
+
+undefined = Undefined.undefined
```

Comment:
Why not just set it to ellipsis? 

```suggestion
undefined = ...
```

Reply:
Cannot use Ellipsis in Union type

Reply:
Why? 

```python
from typing import TYPE_CHECKING, Union

if TYPE_CHECKING:
    ELLIPSIS = ellipsis


a: "Union[str, ELLIPSIS]" = "meme"
```

`ellipsis` builtin only available during type_checking. You can use it to remember it somewhere for later use. The only thing here is that this type hint should be a string, because ELLIPSIS is not available at runtime .

Reply:
Done